mlflow/models/model.py

```diff
@@ -29,55 +30,141 @@
 _MLFLOW_VERSION_KEY = "mlflow_version"
 
 
-class ModelInfo(NamedTuple):
+class ModelInfo:
     """
     The metadata of a logged MLflow Model.
     """
 
-    #: Run relative path identifying the logged model.
-    artifact_path: str
+    def __init__(
+        self,
+        artifact_path: str = None,
+        flavors: Dict[str, Any] = None,
+        model_uri: str = None,
+        model_uuid: str = None,
+        run_id: str = None,
+        saved_input_example_info: Optional[Dict[str, Any]] = None,
+        signature_dict: Optional[Dict[str, Any]] = None,
+        signature=None,  # Optional[ModelSignature]
+        utc_time_created: str = None,
+        mlflow_version: str = None,
+    ):
+        self._artifact_path = artifact_path
+        self._flavors = flavors
+        self._model_uri = model_uri
+        self._model_uuid = model_uuid
+        self._run_id = run_id
+        self._saved_input_example_info = saved_input_example_info
+        self.signature_dict = signature_dict
```

Comment:
This line triggers the deprecation warning and users see the warning every time they log a model with `mlflow.xxx.log_model`.

Reply:
emmm yes, you're right this is a little bit confusing as they're not explicitly calling this. Perhaps we just preserve the warnings inside of setter & getter